### ISSUE TITLE
`maxTtlIfError` behavior clarification.

### ISSUE THREAD

**Issue by JustinTRoss** [StarpTech/apollo-datasource-http#27]
This line makes it appear `maxTtlIfError` is whatever value is provided in seconds for `maxTtlIfError`
https://github.com/StarpTech/apollo-datasource-http/blob/5f0036f6510790e42c5ecf36900367bb6400514f/README.md?plain=1#L113

This line makes it appear as though `maxTtlIfError` actually ends up being `maxTtl + maxTtlIfError`.
https://github.com/StarpTech/apollo-datasource-http/blob/5f0036f6510790e42c5ecf36900367bb6400514f/src/http-data-source.ts#L328

I'm happy to PR clarification once I have it. Just need to know what that is 🙃 .

**Comment by StarpTech** [comment]
Hi @JustinTRoss, you're right this is confusing. The TTL of the error cache must be always longer than the default TTL cache.
`maxTtlIfError` defines the exact duration in seconds of the stale cache. We should update docs with just e.g.

```
maxTtl: 60 // 1min cache
maxTtlIfError: 60 // additional 60s error cache after maxTtl has expired
```

**Comment by JustinTRoss** [comment]
Got it. To confirm:

```
requestCache: {
        maxTtl: 10 * 60, // 10min, will respond for 10min with the result in cache (revalidated every 10min)
        maxTtlIfError: 30 * 60, // 30min, will respond with the result in cache for a further 30 minutes when revalidation attempts result in error (40 minutes total)
},
```

**Comment by StarpTech** [comment]
`(40 minutes total)` is confusing. I'd remove it.

**Comment by JustinTRoss** [comment]
Agreed. It could be better worded. It sounds like the sentiment is generally true though, that the cached value can be returned for up to 40 minutes. This seems necessary for folks to understand. I'll work on conveying it clearly and send a PR with updated docs.

Thanks again! 🙏

**Comment by StarpTech** [comment]
Thanks!
